File: code/go/ch09/src/luago/state/api_push.go
package state

import . "luago/api"

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushnil
func (self *luaState) PushNil() {
	self.stack.push(nil)
}

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushboolean
func (self *luaState) PushBoolean(b bool) {
	self.stack.push(b)
}

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushinteger
func (self *luaState) PushInteger(n int64) {
	self.stack.push(n)
}

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushnumber
func (self *luaState) PushNumber(n float64) {
	self.stack.push(n)
}

// [-0, +1, m]
// http://www.lua.org/manual/5.3/manual.html#lua_pushstring
func (self *luaState) PushString(s string) {
	self.stack.push(s)
}

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushcfunction
func (self *luaState) PushGoFunction(f GoFunction) {
	self.stack.push(newGoClosure(f))
}

// [-0, +1, –]
// http://www.lua.org/manual/5.3/manual.html#lua_pushglobaltable
func (self *luaState) PushGlobalTable() {
	global := self.registry.get(LUA_RIDX_GLOBALS)
	self.stack.push(global)
}

File: code/csharp/ch09/luavm/state/ApiPush.cs
using luavm.api;

namespace luavm.state
{
    public partial struct LuaState
    {
        public void PushNil()
        {
            stack.push(null);
        }

        public void PushBoolean(bool b)
        {
            stack.push(b);
        }

        public void PushInteger(long n)
        {
            stack.push(n);
        }

        public void PushNumber(double n)
        {
            stack.push(n);
        }

        public void PushString(string s)
        {
            stack.push(s);
        }

        public void PushCsharpFunction(CsharpFunction f)
        {
            stack.push(Closure.newCsharpClosure(f));
        }

        public void PushGlobalTable()
        {
            var global = registry.get(Consts.LUA_RIDX_GLOBALS);
            stack.push(global);
        }
    }
}